packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -991,3 +992,68 @@ export function $clientName(
 export function getClientNameOverride(context: TCGCContext, entity: Type): string | undefined {
   return getScopedDecoratorData(context, clientNameKey, entity);
 }
+
+const overrideClientMethodKey = createStateSymbol("overrideClientMethod");
+
+// Recursive function to collect parameter names
+function collectParamNames(properties: RekeyableMap<string, ModelProperty>): string[] {
+  let paramNames: string[] = [];
+
+  properties.forEach((value, key) => {
+    // If the property is of type 'model', recurse into its properties
+    if (value.type.kind === "Model") {
+      paramNames = paramNames.concat(collectParamNames(value.type.properties));
+    } else {
+      paramNames.push(key);
+    }
+  });
+
+  return paramNames;
```

Comment:
using name for validation is not safe. i suggest to use model property instance to do the validation.